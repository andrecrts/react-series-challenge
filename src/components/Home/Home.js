import React, { Component } from 'react'
import { connect } from 'react-redux'
import contentActions from '../../actions/content'
import Header from '../Header'
import Searchbar from '../Searchbar'

const {
  startLoading,
} = contentActions.actions

class Home extends Component {

  componentDidMount() {
    this.props.dispatch(startLoading())
  }

  render() {
    return(
      <div>
        <Header/>
        <Searchbar/>
        Home
      </div>
    )
  }
}

export default connect(state => state)(Home)
