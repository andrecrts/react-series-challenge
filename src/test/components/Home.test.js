import React from 'react';
import Home from '../../components/Home'
import configureStore from 'redux-mock-store'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import 'jest-styled-components'

const middlewares = []
const mockStore = configureStore(middlewares)

test('Home snapshot', () => {
  const initialState = {
    content:{
      data: {data: {title:'title1', images:{original:{webp:'srcImage'}}}},
      favorites:{}
    }
  }
  const store = mockStore(initialState)
  const component = shallow(
    <Home store={store}/>,
  );
  expect(toJson(component)).toMatchSnapshot();
})


