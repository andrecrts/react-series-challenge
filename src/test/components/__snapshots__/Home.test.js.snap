// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home snapshot 1`] = `
<Home
  addToFavContent={[Function]}
  content={
    Object {
      "data": Object {
        "data": Object {
          "images": Object {
            "original": Object {
              "webp": "srcImage",
            },
          },
          "title": "title1",
        },
      },
      "favorites": Object {},
    }
  }
  fetchData={[Function]}
  removeFavContent={[Function]}
  startSearch={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;
